x-pack/plugins/security_solution/public/hosts/components/host_risk_score_table/severity_filter_group.tsx

```diff
@@ -0,0 +1,128 @@
+
+interface SeverityItems {
+  risk: HostRiskSeverity;
+  count: number;
+  checked?: FilterChecked;
+}
+export const SeverityFilterGroup: React.FC<{
+  severityCount: SeverityCount;
+  type: hostsModel.HostsType;
+}> = ({ severityCount, type }) => {
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+
+  const dispatch = useDispatch();
+
+  const onButtonClick = useCallback(() => {
+    setIsPopoverOpen(!isPopoverOpen);
+  }, [isPopoverOpen]);
+
+  const closePopover = useCallback(() => {
+    setIsPopoverOpen(false);
+  }, []);
+
+  const filterGroupPopoverId = useGeneratedHtmlId({
+    prefix: 'filterGroupPopover',
+  });
+  const getHostRiskScoreFilterQuerySelector = useMemo(
+    () => hostsSelectors.hostRiskScoreSeverityFilterSelector(),
+    []
+  );
+  const severitySelectionRedux = useDeepEqualSelector((state: State) =>
+    getHostRiskScoreFilterQuerySelector(state, type)
+  );
+
+  const items: SeverityItems[] = useMemo(() => {
+    const checked: FilterChecked = 'on';
+    return (Object.keys(severityCount) as HostRiskSeverity[]).map((k) => ({
+      risk: k,
+      count: severityCount[k],
+      checked: severitySelectionRedux.includes(k) ? checked : undefined,
+    }));
+  }, [severityCount, severitySelectionRedux]);
+
+  const updateSeverityFilter = useCallback(
+    (selectedSeverity: HostRiskSeverity) => {
+      const currentSelection = severitySelectionRedux ?? [];
+      const newSelection = currentSelection.includes(selectedSeverity)
+        ? currentSelection.filter((s) => s !== selectedSeverity)
+        : [...currentSelection, selectedSeverity];
+      dispatch(
+        hostsActions.updateHostRiskScoreSeverityFilter({
+          severitySelection: newSelection,
+          hostsType: type,
+        })
+      );
+    },
+    [dispatch, severitySelectionRedux, type]
+  );
+
+  const totalActiveHosts = useMemo(
+    () => items.reduce((total, item) => (item.checked === 'on' ? total + item.count : total), 0),
+    [items]
+  );
+
+  const button = useMemo(
+    () => (
+      <EuiFilterButton
+        data-test-subj="host-risk-filter-button"
+        hasActiveFilters={!!items.find((item) => item.checked === 'on')}
+        iconType="arrowDown"
+        isSelected={isPopoverOpen}
+        numActiveFilters={totalActiveHosts}
+        onClick={onButtonClick}
+      >
+        {i18n.HOST_RISK}
+      </EuiFilterButton>
+    ),
+    [isPopoverOpen, items, onButtonClick, totalActiveHosts]
+  );
+
+  return (
+    <EuiFilterGroup>
+      <EuiPopover
+        id={filterGroupPopoverId}
+        button={button}
+        isOpen={isPopoverOpen}
+        closePopover={closePopover}
+        panelPaddingSize="none"
+      >
+        <div className="euiFilterSelect__items">
+          {items.map((item, index) => (
+            <EuiFilterSelectItem
+              data-test-subj={`host-risk-filter-item-${item.risk}`}
+              checked={item.checked}
+              key={index + item.risk}
```

Comment:
isn't `key={index}` enough? just curious

Reply:
Index as a key is an anti-pattern https://robinpokorny.medium.com/index-as-a-key-is-an-anti-pattern-e0349aece318

Reply:
hmmm yes, that's right for dynamic lists, but this list is static, we are always showing the same items in the same order, so `key={index}` would work. Moreover, if we are not going to trust the `index`, why are we prepending it instead of doing just `key={item.risk}`?

Anyway, despite that, I realized the `items` array comes from an `Object.keys(severityCount)` call, so the order is not guaranteed here. In this particular case, as the keys are "non-integer strings", modern browsers will [keep the insertion order](https://www.stefanjudis.com/today-i-learned/property-order-is-predictable-in-javascript-objects-since-es2015/), so it will work while we set the keys always in the proper order. Anyhow, I would not say that relying on the `Object.keys` order for the menu items is a good practice either.

In conclusion, it seems everything will work as it is, but I think it would be better if we generated the menu `items` by iterating over a constant `Array<HostRiskSeverity>`, instead of using the `severityCount` object keys.

Reply:
@semd Good point. I think that the same comment applies to `Object.keys(HOST_RISK_SEVERITY_COLOUR)`. The order of elements in the UI is determined by the order that fields are defined on `HOST_RISK_SEVERITY_COLOUR`.